netket/sampler/abstract_sampler.py

```diff
@@ -27,11 +27,11 @@ def reset(self, init_random=False):
         pass
 
     @property
-    def machine_func(self):
-        return lambda x, out=None: _np.square(_np.absolute(x), out)
+    def machine_pow(self):
+        return 2.0
 
-    @machine_func.setter
-    def machine_func(self, func):
+    @machine_pow.setter
+    def machine_pow(self, m_power):
         raise NotImplementedError
```

Comment:
Ah, maybe it's because there might samplers that work only for pow=2, or pow=5 and you can't change it, I guess?

But then I still think it would be better to have it as a hidden property and define

```
@property 
def machine_pow(self):
 return self._machine_pow

@machine_pow.setter
def machine_pow(self, m_power):
         raise NotImplementedError

Reply:
yes by default samplers sample the square of the machine, and that needs to be implemented in all samplers. Other samplers might implement other powers... Yes, maybe you are right this is a better strategy, raising NotImplementedError

Reply:
Even though I don't understand the difference between  my current implementation 

```python
    @property
    def machine_pow(self):
        return 2

    @machine_pow.setter
    def machine_pow(self, m_power):
        raise NotImplementedError
```

and something like 

```python
    @property
    def machine_pow(self):
        return self._machine_pow

    @machine_pow.setter
    def machine_pow(self, m_power):
        raise NotImplementedError
```

where self._machine_pow=2.0 is an inaccesible property of the abstract class defined once for all in the __init__ of the abstract class. I think it's exactly the same?

Reply:
It's because I was thinking that a sampler could in Principle be defined only for pow≠1, but that seems unlikely in our case, so, I'm fine with your implementation.